src/components/inputs/radio-input/radio-group.js

```diff
@@ -36,25 +52,49 @@ class Group extends React.PureComponent {
   }
 
   render() {
-    const DirectionWrapper =
-      this.props.direction === 'stack' ? Spacings.Stack : Spacings.Inline;
+    const optionElements = React.Children.map(
+      this.props.children,
+      (child, index) => {
+        // NOTE: Allowing to intersperse other elements than `Option`.
+        if (child && child.type.displayName === Option.displayName) {
+          return React.cloneElement(child, {
+            id: `${this.props.id}-${index}`,
+            name: this.props.name,
+            isChecked: this.props.value === child.props.value,
+            isDisabled: child.props.isDisabled || this.props.isDisabled,
+            isReadOnly: this.props.isReadOnly,
+            hasError: this.props.hasError,
+            hasWarning: this.props.hasWarning,
+            onChange: this.props.onChange,
+            onFocus: this.props.onFocus,
+            onBlur: this.props.onBlur,
+          });
+        }
+        return child;
+      }
+    );
+    if (this.props.direction === 'inline') {
+      return (
+        <div id={this.props.id}>
+          <Spacings.Inline
+            {...this.props.directionProps}
+            {...filterDataAttributes(this.props)}
+          >
+            {optionElements}
+          </Spacings.Inline>
+        </div>
+      );
+    }
     return (
-      <div
-        className={this.props.className}
-        {...filterDataAttributes(this.props)}
-      >
-        <DirectionWrapper scale={this.props.scale}>
-          {React.Children.map(this.props.children, child => {
-            // NOTE: Allowing to intersperse other elements than `Option`.
-            if (child && child.type.displayName === Option.displayName)
-              return React.cloneElement(child, {
-                isChecked: this.props.value === child.props.value,
-                name: this.props.name,
-                onChange: this.props.onChange,
-              });
-            return child;
-          })}
-        </DirectionWrapper>
+      <div id={this.props.id}>
+        <Constraints.Horizontal constraint={this.props.horizontalConstraint}>
```

Comment:
Only the stack mode has the horizontal constraint, as it does not make sense to have it in the inline mode.